packages/grid/x-data-grid/src/tests/selection.DataGrid.test.tsx

```diff
@@ -83,6 +83,48 @@ describe('<DataGrid /> - Selection', () => {
     });
   });
 
+  describe('props: checkboxSelection = false (single selection), with keyboard events', () => {
+    it('should select one row at a time on Shift + Space', () => {
+      render(<TestDataGridSelection />);
+      getCell(0, 0).focus();
+      fireEvent.keyDown(document.activeElement || document.body, { key: ' ', shiftKey: true });
+      expect(getSelectedRowIds()).to.deep.equal([0]);
+
+      fireEvent.keyDown(document.activeElement || document.body, { key: 'ArrowDown' });
+      fireEvent.keyDown(document.activeElement || document.body, { key: ' ', shiftKey: true });
+
+      expect(getSelectedRowIds()).to.deep.equal([1]);
+    });
+
+    it(`should not deselect the selected row on Shift + Space`, () => {
```

Comment:
It should. At least, this is how https://mui.com/components/data-grid/selection/#single-row-selection is working.